components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -783,36 +783,46 @@ func NewKymaOrchestrationProcessingQueue(ctx context.Context, db storage.BrokerS
 		step     upgrade_kyma.Step
 		cnd      upgrade_kyma.StepCondition
 	}{
+		{
+			weight: 1,
+			step:   upgrade_kyma.NewCreateClusterConfiguration(db.Operations(), db.RuntimeStates(), reconcilerClient),
+			cnd:    upgrade_kyma.ForKyma2,
+		},
 		{
 			weight: 2,
-			step:   upgrade_kyma.NewOverridesFromSecretsAndConfigStep(db.Operations(), runtimeOverrides, runtimeVerConfigurator),
+			step:   upgrade_kyma.NewCheckClusterConfigurationStep(db.Operations(), reconcilerClient, 15*time.Minute),
```

Comment:
why we need this step here?

Reply:
upgradeQueue does not have `stages` implemented like we do have in provisioning process. This step is at the beginning of the process, because in the first iteration of the step executor loop it won't be executed at all:
```go
	if operation.ClusterConfigurationVersion == 0 {
		// upgrade was trigerred in reconciler, no need to call provisioner and create UpgradeRuntimeInput
		// TODO: deal with skipping steps in case of calling reconciler for Kyma 2.0 upgrade - introduce stages
		log.Debugf("Cluster configuration not yet created, skipping")
		return operation, 0, nil
	}
```

`operation.ClusterConfigurationVersion` is 0 by default, and changes after `apply_cluster_configuration.go` step is executed:
```go
	updatedOperation, repeat := s.operationManager.UpdateOperation(operation, func(operation *internal.UpgradeKymaOperation) {
		operation.ClusterConfigurationVersion = state.ConfigurationVersion
	}, log)
```

`apply_cluster_configuration` being the last step in the queue returns some retry value:
```go
	// return some retry value to get back to initialisation step
	return updatedOperation, 5 * time.Second, nil
```

so we get back to the beginning - initilisation step gets skipped because `operation.ClusterConfigurationVersion` is already not equal `0`, same for `Create_Cluster_Configuration` step which leads us to execution of `Check_Cluster_Configuration` step again which starts fetching for the reconcilitation status.